x/programs/rust/wasmlanche_sdk/src/types.rs

```diff
@@ -81,8 +81,12 @@ impl From<i64> for Bytes32 {
     }
 }
 
+/// A trait that represents an argument that can be passed to & from the host.
 pub trait Argument {
     fn as_bytes(&self) -> Cow<'_, [u8]>;
+    fn from_bytes(bytes: &[u8]) -> Self
```

Comment:
This needs to return `Result<Self, _>`. The error type can be changed later, but you know this will always be fallible so it's best to do the work to change it now. 

Calling `unwrap` is great until you've had to go refactor a ton of code that calls `unwrap` everywhere and you think to yourself, "it would have been so much easier to just return a `Result` in the first place". 

I'll leave a suggestion below for how to not get bogged down in dealing with failure modes while you're prototyping.